drivers/scheduler/dcos/dcos.go

```diff
@@ -208,24 +220,124 @@ func (d *dcos) DeleteTasks(ctx *scheduler.Context) error {
 	return nil
 }
 
-func (d *dcos) GetVolumes(ctx *scheduler.Context) ([]string, error) {
-	// TODO: Implement this method
-	return nil, nil
-}
-
 func (d *dcos) GetVolumeParameters(ctx *scheduler.Context) (map[string]map[string]string, error) {
-	// TODO: Implement this method
-	return nil, nil
+	// DC/OS does not have volume objects like Kubernetes. We get the volume information from
+	// the app spec and get the options parsed from the respective volume driver
+	volDriver, err := volume.Get(d.volDriverName)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]map[string]string)
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*marathon.Application); ok {
+			params := *obj.Container.Docker.Parameters
```

Comment:
This will need to be updated when testing with UCR/mesos containers: https://docs.portworx.com/scheduler/mesosphere-dcos/portworx_volumes.html#mesosucr-containers

Reply:
Yes we need to handle mesos containers. Creating a issue to track it.. will soon work on it.